### PR TITLE
Task 5 redux

### PR DESCRIPTION
**Pull request by sixtyXi**

### CODE REVIEW

File: src/services/moviesAPI.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const mapMovie = movie => {
+  return { ...movie, release_date: new Date(movie.release_date) };
+};
+
+const getMovies = (params = {}) => {
+  const query = createQuery(params);
+
+  return fetch(`${BASE_URL}${query}`)
```
Comment (IrinaShilina) [inline src/services/moviesAPI.js]: Error handling for all `fetch` functions would be nice.

File: src/list-movie/components/SortFilterContainer/SortFilterContainer.test.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('SortContainer', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow(<SortFilterContainer filters={SORT_FILTERS} />);
+  });
+
+  test('renders single component *FilterPanel*', () => {
+    expect(wrapper.find(FilterPanel).length).toBe(1);
```
Comment (IrinaShilina) [inline src/list-movie/components/SortFilterContainer/SortFilterContainer.test.jsx]: I think it makes more sense to test component methods. `handleChange` in this case. Also,  `mapStateToProps` and `mapDispatchToProps` may also be tested.

File: src/list-movie/components/SortFilterContainer/SortFilterContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  render() {
+    const { currentFilter } = this.props;
+
+    return (
+      <FilterPanel
+        handleChange={this.handleChange}
+        title={SORT_TITLE}
+        filters={SORT_FILTERS}
+        type={this.panelType}
```
Comment (IrinaShilina) [inline src/list-movie/components/SortFilterContainer/SortFilterContainer.jsx]: Just pass 'panel'.

Comment (sixtyXi) [reply src/list-movie/components/SortFilterContainer/SortFilterContainer.jsx]: Done.

File: src/list-movie/components/SortFilterContainer/SortFilterContainer.test.jsx (hunk 2 above)
Comment (sixtyXi) [reply src/list-movie/components/SortFilterContainer/SortFilterContainer.test.jsx]: All tests for stateful components should be refactored due to using connect. I will try to do it in the next task.

File: src/services/moviesAPI.js (hunk 1 above)
Comment (sixtyXi) [reply src/services/moviesAPI.js]: I use ErrorBoundary to catch all errors. Should I implement additional logic for fetch or error parsing in ErrorBoundary?

File: src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
-    searchTxt: ''
+  handleChange = event => {
+    this.props.setSearchValue(event.target.value);
   };
 
-  handleChange = (event) => {
-    this.setState({ searchTxt: event.target.value });
-  }
+  handleSubmit = event => {
+    if (event.keyCode === 13) {
```
[the remark this answers is not in the document]
Comment (sixtyXi) [reply src/list-movie/components/SearchFieldContainer/SearchFieldContainer.jsx]: This method handles keyDown event. I have renamed it to handleKeyDown.

File: src/containers/MovieThumbContainer/MovieThumbContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { MovieThumb } from '../../components/MovieThumb/MovieThumb';
+import { fetchMovie } from '../../actions/actions';
+
+export const MovieThumbContainer = props => {
+  const { getMovie, movie } = props;
+
+  return <MovieThumb handleClick={() => getMovie(movie.id)} movie={movie} />;
```
[the remark this answers is not in the document]
Comment (sixtyXi) [reply src/containers/MovieThumbContainer/MovieThumbContainer.jsx]: Ups, I did it again ) Thanks! Refactored.

File: src/components/MovieThumb/MovieThumb.jsx [hunk 6]
```diff
@@ -1,24 +1,36 @@
...
+    poster_path: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    genres: PropTypes.arrayOf(PropTypes.string),
+    release_date: PropTypes.instanceOf(Date)
+  }).isRequired,
+  handleClick: PropTypes.func
+};
+
+MovieThumb.defaultProps = {
+  handleClick: () => {}
```
[the remark this answers is not in the document]
Comment (sixtyXi) [reply src/components/MovieThumb/MovieThumb.jsx]: Done.

File: src/actions/actions.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+export const SET_SEARCH = 'SET_SEARCH';
+export const SET_MOVIE = 'SET_MOVIE';
+export const SET_MOVIES_BY_GENRE = 'SET_MOVIES_BY_GENRE';
+
+export const setMovieList = movies => {
+  return {
+    type: SET_MOVIE_LIST,
+    payload: {
+      movies
+    }
```
[the remark this answers is not in the document]
Comment (sixtyXi) [reply src/actions/actions.js]: It's just recommendation from lecture. I will keep in mind, thanks!

File: src/list-movie/components/MovieQtyContainer/MovieQtyContainer.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+export class MovieQtyContainer extends PureComponent {
+  render() {
+    const { movieQty } = this.props;
+
+    return <div>{`${movieQty} movies found`}</div>;
```
Comment (IrinaShilina) [inline src/list-movie/components/MovieQtyContainer/MovieQtyContainer.jsx]: Not a big deal but you can write it just like:
```
return <div>${movieQty} movies found</div>;
```

**Pull request merged**